src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -40,8 +40,10 @@ DeparseAlterTableSchemaStmt(Node *node)
 static void
 AppendAlterTableSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt)
 {
-	Assert(stmt->objectType == OBJECT_TABLE);
-	appendStringInfo(buf, "ALTER TABLE ");
+	Assert(stmt->objectType == OBJECT_TABLE || stmt->objectType == OBJECT_FOREIGN_TABLE);
+
+	bool isForeignTable = stmt->objectType == OBJECT_FOREIGN_TABLE;
+	appendStringInfo(buf, "ALTER %sTABLE ", isForeignTable ? "FOREIGN " : "");
```

Comment:
very minor: I think it'd be easier to read `ALTER %s TABLE ", isForeignTable ? "FOREIGN" : "`. Is that breaking other tests?

Reply:
no that wouldn't break anything but it just annoys me with two space characters together, between `ALTER` and `TABLE`.